pkg/devfile/parser/types.go

```diff
@@ -65,22 +65,46 @@ func (d DevfileObj) OverrideComponents(overridePatch []common.DevfileComponent)
 func (d DevfileObj) OverrideCommands(overridePatch []common.DevfileCommand) error {
 	for _, patchCommand := range overridePatch {
 		found := false
+
 		for _, originalCommand := range d.Data.GetCommands() {
+
 			if strings.ToLower(patchCommand.Id) == originalCommand.Id {
 				found = true
-				var updatedExec common.Exec
 
-				merged, err := handleMerge(originalCommand.Exec, patchCommand.Exec, common.Exec{})
-				if err != nil {
-					return err
-				}
-
-				err = json.Unmarshal(merged, &updatedExec)
-				if err != nil {
-					return errors.Wrap(err, "failed to unmarshal override commands")
+				var devfileCommand common.DevfileCommand
+
```

Comment:
can we break the composite command override into a util function?

Reply:
Updated, thanks @adisky. I've kept the unit tests that I added as part of TestDevfileObj_OverrideCommands, however.